extensions/amp-a4a/amp-a4a-format.md

```diff
@@ -0,0 +1,392 @@
+[general AMP documents do](https://github.com/ampproject/amphtml/blob/master/spec/amp-boilerplate.md):
+
+```
+<style amp-boilerplate>body{visibility:hidden}</style>
+```
+
+_Rationale:_ The `amp-boilerplate` style hides body content until the AMP 
+runtime is ready and can unhide it.  If Javascript is disabled or the AMP 
+runtime fails to load, the default boilerplate ensures that the content is 
+eventually displayed regardless.  In A4A, however, if Javascript is entirely 
+disabled, A4A won't run and no ad will ever be shown, so there is no need for
+the `<noscript>` section.  In the absence of the AMP runtime, most of the 
+machinery that A4A and ad creatives rely on (e.g., analytics for visibility 
+tracking or `amp-img` for content display) won't be available, so it's better to 
+display no ad than a malfunctioning one.
+
+Note that the same rules about mutations to the boilerplate text apply as in 
+the [general AMP boilerplate](https://github.com/ampproject/amphtml/blob/master/spec/amp-boilerplate.md).
+
+### CSS
+
+1. `position:fixed` and `position:sticky` are prohibited in 
+creative CSS.
+
+  _Rationale_: position:fixed breaks out of shadow DOM, which A4A depends on.  Also, Ads in AMP are already not allowed to use fixed position.
+
+1. Creative CSS is limited to 20kb.
+
+  _Rationale_: Large CSS blocks bloat the creative, increase network latency, and degrade page performance.
+
+1. CSS: transition and animation are subject to additional restrictions.
+
+  _Rationale_: AMP must be able to control all animations belonging to an ad, so that it can stop them when the ad is not on screen or system resources are very low.
+
+#### CSS Animations and Transitions
+
+##### Selectors
+
+The `transition` and `animation` properties are only allowed on selectors that:
+- Contain only `transition`, `animation`, `transform`, `visibility`, or 
+  `opacity` properties.
+- Start with `.amp-animate` followed by a space.
+
+  _Rationale:_ This allows the AMP runtime to remove this  class from context
+   to deactivate animations, when necessary for page performance.
+
+**Good**
+```css
+.amp-animate .box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+**Bad**
+
+Property not allowed in CSS class.
+```css
+.amp-animate .box {
+    color: red;  // non-animation property not allowed in animation selector
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+Missing context class `.amp-animate`.
+```css
+.box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+##### Transitionable and animatable properties
+
+The only properties that may be transitioned are opacity and transform.
+([Rationale](http://www.html5rocks.com/en/tutorials/speed/high-performance-animations/))
+
+**Good**
+```css
+transition: transform 2s;
+```
+
+**Bad**
+```css
+transition: background-color 2s;
+```
+
+**Good**
+```css
+@keyframes turn {
```

Comment:
-CSS format proposal to organize animation timeline better
One of the keys difference I find when animating with JS is the control I have in the timeline (when to animate each element, delays, etc), this is kind of lost when only CSS Animations are whitelisted.
We should propose a way to organise the CSS/Keyframes to make it easier to understand the timeline of the ad animation.